assets/js/googlesitekit/widgets/datastore/areas.js

```diff
@@ -0,0 +1,215 @@
+const REGISTER_WIDGET_AREA = 'REGISTER_WIDGET_AREA';
+
+export const INITIAL_STATE = {
+	areas: {},
+	areaWidgets: {},
+	contexts: {
+		analytics: [],
+		dashboard: [],
+	},
+};
+
+export const actions = {
+	/**
+	 * Assigns a widget area to one (or several) contexts.
+	 *
+	 * Accepts an area slug to register as the first argument, then either a string
+	 * (for a single context slug) or array of contexts slugs (to assign the widget area
+	 * to multiple contexts).
+	 *
+	 * Does not error if any of the requested contexts do not exist, but will console.warn
+	 * a message about non-existant contexts.
+	 *
+	 * Does not error if the area slug provided does not exist, but will console.warn a
+	 * message about non-existent areas.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param  {string}         areaSlug     Widget Area's slug.
+	 * @param  {(string|Array)} contextSlugs Widget Context's slug(s).
+	 * @return {Object}                      Redux-style action.
+	 */
+	assignWidgetArea( areaSlug, contextSlugs ) {
+		let contextSlugsAsArray;
+		if ( typeof contextSlugs === 'string' ) {
+			contextSlugsAsArray = [ contextSlugs ];
+		} else {
+			contextSlugsAsArray = [ ...contextSlugsAsArray ];
+		}
+
+		return {
+			payload: { areaSlug, contextSlugs: contextSlugsAsArray },
+			type: ASSIGN_WIDGET_AREA,
+		};
+	},
+
+	/**
+	 * Creates a widget area with a given name and settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param  {string}             areaSlug           Widget Area's slug.
+	 * @param  {Object}             settings           Widget Area's settings.
+	 * @param  {number}             settings.priority  Priority for this widget area. Default: 10.
+	 * @param  {string}             settings.title     Title for this widget area.
+	 * @param  {string}             settings.subtitle  Subtitle for this widget area.
+	 * @param  {(string|undefined)} settings.icon      URL to SVG icon for this widget area.
+	 * @param  {string}             settings.style     Widget area style (one of "boxes", "composite").
+	 * @return {Object}                                Redux-style action.
+	 */
+	registerWidgetArea( areaSlug, settings ) {
+		invariant( areaSlug, 'areaSlug is required.' );
+		invariant( settings, 'settings is required.' );
+		invariant( settings.title, 'settings.title is required.' );
+		invariant( settings.subtitle, 'settings.subtitle is required.' );
+		invariant( settings.style, 'settings.style is required.' );
+
+		const settingsToUse = {
+			priority: 10,
+			...settings,
+		};
+
+		return {
+			payload: { areaSlug, settings: settingsToUse },
+			type: REGISTER_WIDGET_AREA,
+		};
+	},
+};
+
+export const controls = {};
+
+export const reducer = ( state, { type, payload } ) => {
+	switch ( type ) {
+		case ASSIGN_WIDGET_AREA: {
+			const { areaSlug, contextSlugs } = payload;
+
+			const { contexts } = state;
+			contextSlugs.forEach( ( contextSlug ) => {
+				if ( contexts[ contextSlug ] === undefined ) {
+					return;
```

Comment:
This should not return, but create an entry for the context with that slug. Per our definition, contexts are currently implicit. If we want to change that, let's discuss separately, but for now have them created "silently".

Reply:
Ahhhh, okay. Gotcha. Works for me!